Compile: error building go interface

### Describe the bug

Compile fails with

```
make[2]: Entering directory '/.../src/ucx/build/bindings/go'
cd /.../src/ucx/build/../ucx-git/bindings/go/src/examples/perftest ;\
go build -o /.../src/ucx/build/bindings/go/.libs/tmp/goperftest
# examples/perftest
./perftest.go:346:18: flag.CommandLine.Func undefined (type *flag.FlagSet has no field or method Func)
make[2]: *** [Makefile:634: goperftest] Error 2
```

### Setup and versions

ucx @ master ef9b1906d3598f67ba7cfb6e0894b3882e7e5db1

go version go1.13.3 linux/arm64 (from CentOS)

- OS version (e.g Linux distro) + CPU architecture (x86_64/aarch64/ppc64le/...)
   - `cat /etc/issue` or `cat /etc/redhat-release` + `uname -a`

```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"

$ uname -a
Linux login1 4.18.0-147.el8.aarch64 #1 SMP Wed Dec 4 21:57:21 UTC 2019 aarch64 aarch64 aarch64 GNU/Linux
```

--- a/config/m4/go.m4
+++ b/config/m4/go.m4
@@ -18,7 +18,9 @@ AS_IF([test "x$with_go" != xno],
       [
             AC_CHECK_PROG(GOBIN, go, yes)
             AS_IF([test "x${GOBIN}" = "xyes"],
-                  [go_happy="yes"],
+                  [AS_VERSION_COMPARE([1.16], [`go version | awk '{print substr($3, 3, length($3)-2)}'`],
+                        [go_happy="yes"], [go_happy="yes"],
+                        [AC_MSG_ERROR([Need Go compiler of version 1.16 or newer.])])],
                   [AS_IF([test "x$with_go" = "xguess"],
                         [AC_MSG_WARN([Disabling GO support - GO compiler not found.])],
                         [AC_MSG_ERROR([GO support was explicitly requested, but go compiler not found.])])

